### ISSUE TITLE
`--stdin-batch-closed-world`: a directory implied by supplied paths is reported as a missing link target

### ISSUE THREAD

**Issue by darwin** [rvben/rumdl#911]
## Summary

In closed-world mode a link to a directory is reported by MD057 as "target not in the supplied document set", even when the batch supplies documents inside that directory. On disk the same link passes, because the directory exists.

## Reproduction (rumdl 0.2.76)

```bash
mkdir -p repro/docs && cd repro
printf '# A\n\nSee [the docs folder](docs/) and [the docs page](docs/b.md).\n' > a.md
printf '# B\n' > docs/b.md

rumdl check a.md docs/b.md
# Success: No issues found in 2 files

{ printf 'a.md\0'; cat a.md; printf '\0docs/b.md\0'; cat docs/b.md; printf '\0'; } \
  | rumdl check --stdin-batch --stdin-batch-closed-world
# a.md:3:23: [MD057] Relative link 'docs/' target not in the supplied document set
```

## Expected

A directory counts as existing when at least one supplied path lies under it, i.e. the supplied set implies it, the same way git implies directories from the paths it tracks. The link above should pass in closed-world mode, as it does on disk.

## Impact

On the corpus from #789 (7 575 tracked paths), closed-world mode with every tracked path supplied reported 2 280 findings, and **all 2 280 were links to directories**. There were no other differences from the on-disk run. We filter them on the caller side for now.

---

🤖 Drafted with [Claude Code](https://claude.com/claude-code). The reproduction and measurements were run locally.
